flang/lib/Optimizer/Builder/HLFIRTools.cpp

```diff
@@ -967,6 +969,15 @@ hlfir::LoopNest hlfir::genLoopNest(mlir::Location loc,
       auto ub = builder.createConvert(loc, indexType, extent);
       auto doLoop =
           builder.create<fir::DoLoopOp>(loc, one, ub, one, isUnordered);
+      if (!couldVectorize) {
+        mlir::LLVM::LoopVectorizeAttr va{mlir::LLVM::LoopVectorizeAttr::get(
+            builder.getContext(),
+            /*disable=*/builder.getBoolAttr(true), {}, {}, {}, {}, {}, {})};
+        mlir::LLVM::LoopAnnotationAttr la = mlir::LLVM::LoopAnnotationAttr::get(
+            builder.getContext(), {}, /*vectorize=*/va, {}, /*unroll*/ {},
+            /*unroll_and_jam*/ {}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {});
+        doLoop.setLoopAnnotationAttr(la);
+      }
```

Comment:
Please could you add a test verifying that this attribute makes it through lowering to the right place in LLVMIR (or if there is already a test showing a fir.do_loop with this attribute getting lowered to LLVMIR then that's okay).

Reply:
There's [this](https://github.com/llvm/llvm-project/blob/7efb79b7053cec7cd712c0fede14e1478f353917/flang/test/Fir/vector-always.fir) test, it checks the `disable=false` case rather than `disable=true` but it does test the lowering. I can still add a copy like `vector-never.fir` if you'd like me to.